src/pipecat/processors/aggregators/llm_response_universal.py

```diff
@@ -443,101 +375,52 @@ async def _handle_transcription(self, frame: TranscriptionFrame):
-                    )
-                    self._emulating_vad = False
-            finally:
-                self._aggregation_event.clear()
-
-    async def _maybe_emulate_user_speaking(self):
-        """Maybe emulate user speaking based on transcription.
-
-        Emulate user speaking if we got a transcription but it was not
-        detected by VAD. Behavior when bot is speaking depends on the
-        enable_emulated_vad_interruptions parameter.
-        """
-        # Check if we received a transcription but VAD was not able to detect
-        # voice (e.g. when you whisper a short utterance). In that case, we need
-        # to emulate VAD (i.e. user start/stopped speaking), but we do it only
-        # if the bot is not speaking. If the bot is speaking and we really have
-        # a short utterance we don't really want to interrupt the bot.
-        if (
-            not self._user_speaking
-            and not self._waiting_for_aggregation
-            and len(self._aggregation) > 0
-        ):
-            if self._bot_speaking and not self._params.enable_emulated_vad_interruptions:
-                # If emulated VAD interruptions are disabled and bot is speaking, ignore
-                logger.debug("Ignoring user speaking emulation, bot is speaking.")
-                await self.reset()
-            else:
-                # Either bot is not speaking, or emulated VAD interruptions are enabled
-                # - trigger user speaking emulation.
-                await self.push_frame(EmulateUserStartedSpeakingFrame(), FrameDirection.UPSTREAM)
-                self._emulating_vad = True
+
+    async def _on_user_turn_started(self, strategy: BaseUserTurnStartStrategy):
+        await self._trigger_user_turn_start(strategy)
+
+    async def _on_bot_turn_started(self, strategy: BaseBotTurnStartStrategy):
+        await self._trigger_bot_turn_start(strategy)
+
+    async def _on_push_frame(
+        self,
+        strategy: BaseUserTurnStartStrategy | BaseBotTurnStartStrategy,
+        frame: Frame,
+        direction: FrameDirection = FrameDirection.DOWNSTREAM,
+    ):
+        await self.push_frame(frame, direction)
+
+    async def _on_broadcast_frame(
+        self,
+        strategy: BaseUserTurnStartStrategy | BaseBotTurnStartStrategy,
+        frame_cls: Type[Frame],
+        **kwargs,
+    ):
+        await self.broadcast_frame(frame_cls, **kwargs)
+
+    async def _trigger_user_turn_start(self, strategy: BaseUserTurnStartStrategy):
+        # Reset all user turn start strategies to start fresh.
+        if self.turn_start_strategies:
+            for s in self.turn_start_strategies.user:
+                await s.reset()
+
+        if self._params.enable_user_speaking_frames:
+            logger.debug(f"User started speaking (user turn start strategy: {strategy})")
+            await self.broadcast_frame(UserStartedSpeakingFrame, emulated=strategy is None)
+            await self.push_frame(InterruptionFrame())
```

Comment:
Do we want to push the interruption from this processor, or do we want to send it the same way as `UserStartedSpeakingFrame`, above?

Reply:
Done